docs/howto/overriding-shell.txt

```diff
@@ -0,0 +1,49 @@
+=========================
+How to Override the Shell
+=========================
+
+Now the shell imports your application's models for you. Furthermore, you might want either to not import models or import extra things in the shell.
+
+Setting up the App Directory
+============================
+
+Following the :doc:`guide </howto/custom-management-commands>` on how to create a custom management command, you can set up your app directory like this:
+
+.. code-block:: text
+
+    polls/
+        __init__.py
+        models.py
+        management/
+            __init__.py
+            commands/
+                __init__.py
+                shell.py
+        tests.py
+        views.py
+
+In this way, putting the app before the overridden shell app in :setting:`INSTALLED_APPS`, you will override the default shell command.
+
+Overriding the Shell
+====================
+
+To override the command, you have to override the ``shell.Command.get_namespace`` in order to return a namespace that maps every name to the object imported. For example, you can edit ``polls/management/commands/shell.py`` to look like this::
+
+    from django.core.management.commands import shell
+
+
+    class Command(shell.Command):
+        def get_namespace(self):
+            from django.urls.base import resolve, reverse
+
+            return {
+                **super().get_namespace(),
+                "resolve": resolve,
+                "reverse": reverse,
+            }
+
+This way, your overridden shell will import models from your apps and also the ``resolve`` and ``reverse`` methods so that you can use those without import.
+
+.. note::
+
+    If you want your shell not to import models, you can also omit calling the ``**super().get_namespace()`` method when returning the namespace because it can be empty.
```

Comment:
```suggestion
    If you prefer your shell without any auto-imported model, you can omit calling the ``**super().get_namespace()`` method when returning the namespace, as it can be empty.
```

Reply:
not sure if this note is needed, but maybe a nice-to-have.